backend/src/services/automationService.ts

```diff
@@ -0,0 +1,106 @@
+import {
+  AutomationCriteria,
+  AutomationData,
+  AutomationState,
+  CreateAutomationRequest,
+  UpdateAutomationRequest,
+} from '../types/automationTypes'
+import { IServiceOptions } from './IServiceOptions'
+import SequelizeRepository from '../database/repositories/sequelizeRepository'
+import AutomationRepository from '../database/repositories/automationRepository'
+import { PageData } from '../types/common'
+import { ServiceBase } from './serviceBase'
+
+export default class AutomationService extends ServiceBase<
+  AutomationData,
+  string,
+  CreateAutomationRequest,
+  UpdateAutomationRequest,
+  AutomationCriteria
+> {
+  public constructor(options: IServiceOptions) {
+    super(options)
+  }
+
+  /**
+   * Creates a new active automation
+   * @param req {CreateAutomationRequest} data used to create a new automation
+   * @returns {AutomationData} object for frontend to use
+   */
+  override async create(req: CreateAutomationRequest): Promise<AutomationData> {
+    const transaction = await SequelizeRepository.createTransaction(this.options.database)
+
+    try {
+      // create an active automation
+      const result = await new AutomationRepository(this.options).create({
+        ...req,
+        state: AutomationState.ACTIVE,
```

Comment:
Should automations always be created with a default `active` state?
Also, if state is boolean (`active`/`disabled`) I wonder if we should change it for a proper boolean like `published` or something similar

Reply:
In the future the state could also be `error` if an automation fails to be performed after 5 tries or something like that - this is why `state` column was used. And AFAIK the default state to be `active` is how normally things are created - from a users perspective you can still deactivate it immediately after if you want.

Reply:
I see, makes sense then 👍 
As for the default state, at least on [make.com](https://make.com) automation flows are always created as a draft/disabled to prevent users from triggering immediately the flow without knowing — maybe we could ask Jonathan/Nuno just to be sure?